lib/perl/Genome/Model/Tools/Manta/Base.pm

```diff
@@ -0,0 +1,246 @@
+package Genome::Model::Tools::Manta::Base;
+
+use strict;
+use warnings;
+
+use Genome;
+use Carp qw(confess);
+
+class Genome::Model::Tools::Manta::Base {
+    is => 'Command',
+    is_abstract => 1,
+    attributes_have => [
+        tool_arg_name => {
+            is => 'String',
+            is_optional => 1,
+        },
+        tool_input_file => {
+            is => 'Boolean',
+            is_optional => 1,
+        },
+    ],
+    has_param => [
+        version => {
+            is => 'Version',
+            doc => 'The version of Manta to use.',
+            valid_values => ['0.29.6'],
+        },
+    ],
+    has_optional_output => [
+        output_files => {
+            is_many => 1,
+        },
+        input_files => {
+            is_many => 1,
+        },
+    ],
+};
+
+sub help_detail {
+    "Manta calls structural variants (SVs) and indels from mapped paired-end sequencing reads."
+}
+
+my %TOOL_VERSIONS = (
+    '0.29.6'     => '/gscmnt/gc13001/info/model_data/jwalker_scratch/src/manta-0.29.6.centos5_x86_64/bin',
```

Comment:
> On Apr 20, 2016, at 2:10 PM, Thomas B. Mooney <notifications@github.com> wrote:
>
> In lib/perl/Genome/Model/Tools/Manta/Base.pm:
>
> > +    has_optional_output => [
> > +        output_files => {
> > +            is_many => 1,
> > +        },
> > +        input_files => {
> > +            is_many => 1,
> > +        },
> > +    ],
> > +};
> > +
> > +sub help_detail {
> > +    "Manta calls structural variants (SVs) and indels from mapped paired-end sequencing reads."
> > +}
> > +
> > +my %TOOL_VERSIONS = (
> > +    '0.29.6'     => '/gscmnt/gc13001/info/model_data/jwalker_scratch/src/manta-0.29.6.centos5_x86_64/bin',
>
> Well, it sounds like we're not allowed to do this in a good way anymore... so carry on!

Can you make paths to executables configurable?

my %TOOL_VERSIONS = (
    '0.29.6'     => ${GENOME_TOOL_EXEC_PATH_MANTA}

/etc/profile.d/genome.sh:
GENOME_TOOL_EXEC_PATH_MANTA=/gscmnt/gc13001/info/model_data/jwalker_scratch/src/manta-0.29.6.centos5_x86_64/bin

OR:

/etc/genome/config.yaml:
GENOME_TOOL_EXEC_PATH_MANTA: /gscmnt/gc13001/info/model_data/jwalker_scratch/src/manta-0.29.6.centos5_x86_64/bin





________________________________
The materials in this message are private and may contain Protected Healthcare Information or other information of a sensitive nature. If you are not the intended recipient, be advised that any unauthorized use, disclosure, copying or the taking of any action in reliance on the contents of this information is strictly prohibited. If you have received this email in error, please immediately notify the sender via telephone or return mail.